### PR TITLE
Fix URL encoding on various android/java versions

### LINKED ISSUE #3003: Pdf preview doesn't work on MessageListView
**Describe the bug**
When user clicks on pdf file inside io.getstream.chat.android.ui.message.list.MessageListView, then "No preview available".

**SDK version**
//stream chat deps
    implementation "io.getstream:stream-chat-android-ui-components:4.27.0"
    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:2.4.0"
    implementation "io.coil-kt:coil:1.4.0"
    implementation 'androidx.activity:activity-ktx:1.4.0'

**To Reproduce**
1. Go to list of message inside MessageListView.
2. Upload pdf file.
3. Click on the pdf file.
4. See "No preview available" screen.

**Expected behavior**
See pdf file content preview.

**Device:**
 -Android emulator api 31
 - OnePlus 5t (android 10.0.1), 
 - OnePlus 6t.

**Screenshots**
"No preview available" screen
image

Chat with pdf where user clicks to open pdf preview
image

I use this code to present the list of message, where I click to the PDF file in the chat.
```
<io.getstream.chat.android.ui.message.list.MessageListView
        android:id="@+id/messageListView"
        android:layout_width="0dp"
        android:layout_height="0dp"
        android:clipToPadding="false"
        android:paddingBottom="16dp"
        app:layout_constraintBottom_toTopOf="@+id/messageInputView"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toBottomOf="@+id/headerDividerView" />
```

### PR DESCRIPTION
**Pull request by filbabic**
Closes #3003 

### 🎯 Goal

After researching into a user-reported issue, we've found that our encoding logic wasn't working as the property used for the encoding was faulty.

### 🛠 Implementation details

Instead of using `encoding.toString()` we used `encoding.name()` as in different Java/JVM environments (e.g. older devices) `toString()` will not return the proper value.

### 🧪 Testing

1. Run the UI comp sample app on an API 21-25 device _without_ this fix and attempt to open a PDF file. The app will crash or it will throw an error saying it cannot preview a PDF.
2. Run the sample app _with_ this fix and it should work fine.

#### General
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

giphy (1)

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/view/activity/AttachmentDocumentActivity.java [hunk 1]
```diff
@@ -65,17 +66,23 @@ private void configUIs() {
...
+    /**
+     * Encodes the give URL in a format that can be opened in the browser.
+     *
+     * @param url The URL of the file to load
+     * @return String representation of the encoded URL.
+     */
     private String encodeUrl(String url) {
         try {
-            return URLEncoder.encode(url, StandardCharsets.UTF_8.toString());
+            return URLEncoder.encode(url, StandardCharsets.UTF_8.name());
```
Comment (filbabic) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/view/activity/AttachmentDocumentActivity.java:77]: Alternatively, we could use `Charset.defaultCharset().name()` here, but for Android, it's always `UTF-8` so this should be fine.

**Comment by bychkovdmitry** [comment]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by bychkovdmitry — APPROVED** [review]
Tested on emulator API 23 and the solution fixes the crash. Worth updating the change log.

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/view/activity/AttachmentDocumentActivity.java [hunk 2]
```diff
@@ -65,17 +66,23 @@ private void configUIs() {
         webView.setWebViewClient(new AppWebViewClients());
     }
 
+    /**
+     * Encodes the give URL in a format that can be opened in the browser.
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/view/activity/AttachmentDocumentActivity.java:70]:
```suggestion
     * Encodes the given URL in a format that can be opened in the browser.
```

Comment (filbabic) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/view/activity/AttachmentDocumentActivity.java:70]: Will update with the changelog!

**Pull request merged**